src/mlpack/tests/one_hot_encoding_test.cpp

```diff
@@ -0,0 +1,164 @@
+    BOOST_REQUIRE_EQUAL(matrix.at(i), output.at(i));
+}
+
+BOOST_AUTO_TEST_CASE(OneHotEncodingInputTest)
+{
+  arma::Mat<int> matrix;
+  arma::Mat<int> input;
+  input = "1 1 -1 -1 -1 -1 1 1;"
+           "-1 1 -1 -1 -1 -1 1 -1;";
+  matrix = "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;";
+
+  // Output matrix to save onehotencoding results.
+  arma::Mat<int> output;
+  arma::ucolvec indices("1");
+  data::OneHotEncoding(input, indices, output);
+
+  BOOST_REQUIRE_EQUAL(matrix.n_cols, output.n_cols);
+  BOOST_REQUIRE_EQUAL(matrix.n_rows, output.n_rows);
+  for (size_t i = 0; i < output.n_elem; i++)
+    BOOST_REQUIRE_EQUAL(matrix.at(i), output.at(i));
+}
+
+BOOST_AUTO_TEST_CASE(OneHotEncodingBigInputTest)
+{
+  arma::Mat<int> matrix;
+  arma::Mat<int> input;
+  input = "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;";
+
+  matrix = "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;";
+
+  // Output matrix to save one-hot encoding results.
+  arma::Mat<int> output;
+  arma::ucolvec indices("1 3");
+  data::OneHotEncoding(input, indices, output);
+
+  BOOST_REQUIRE_EQUAL(matrix.n_cols, output.n_cols);
+  BOOST_REQUIRE_EQUAL(matrix.n_rows, output.n_rows);
+  for (size_t i = 0; i < output.n_elem; i++)
+    BOOST_REQUIRE_EQUAL(matrix.at(i), output.at(i));
+}
+
+BOOST_AUTO_TEST_CASE(OneHotEncodingVeryBigInputTest)
+{
+  arma::Mat<int> matrix;
+  arma::Mat<int> input;
+  input = "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "-1 1 -1 -1 -1 -1 1 -1;"
+          "1 1 -1 -1 -1 -1 1 1;"
+          "1 1 -1 -1 -1 -1 1 1;";
+
+  matrix = "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 0 1 1 1 1 0 1;"
+           "0 1 0 0 0 0 1 0;"
+           "1 1 -1 -1 -1 -1 1 1;"
+           "1 1 -1 -1 -1 -1 1 1;";
+
+  // Output matrix to save one-hot encoding results.
+  arma::Mat<int> output;
+  arma::ucolvec indices("1 4 7 8");
+  data::OneHotEncoding(input, indices, output);
+
+  BOOST_REQUIRE_EQUAL(matrix.n_cols, output.n_cols);
+  BOOST_REQUIRE_EQUAL(matrix.n_rows, output.n_rows);
+  for (size_t i = 0; i < output.n_elem; i++)
+    BOOST_REQUIRE_EQUAL(matrix.at(i), output.at(i));
+}
```

Comment:
Could you add a test case where there are more than two categories in an input?  Consider a test case where you randomly generate a matrix with integer values in the categorical columns between, e.g., 1 and 10.  Consider also, instead of hard-coding the inputs and the outputs, programmatically checking each value.  For instance, you can ensure that for a given `input(row, col)` that the corresponds rows of `output(..., col)` are correct.

Reply:
Line 164 - 209 

Test has 3 categorical data in the input

Reply:
Ahhh, you're right, thank you. :+1: